src/relational.cpp

```diff
@@ -60,11 +60,11 @@ external_pointer<T> make_external_prot(const string &rclass, SEXP prot, ARGS &&.
 	return make_external<ColumnRefExpression>("duckdb_expr", names);
 }
 
-[[cpp11::register]] SEXP rapi_expr_constant(sexp val) {
+[[cpp11::register]] SEXP rapi_expr_constant(sexp val, bool typed_logical_null) {
```

Comment:
I believe this should instead not have the `typed_logical_null` argument, handle `val == R_NilValue` upfront, and use `RApiTypes::SexpToValue(val, 0, true)`

Reply:
How so? I just changed this to `typed_logical_null := false` to support a special case in duckdb. I need to better understand the original motivation.